src/qibolab/pulses.py

```diff
@@ -660,429 +672,80 @@ def __repr__(self):
-            return self._finish.value
-        return self._finish
-
-    @property
-    def se_start(self) -> se_int:
-        """Returns a symbolic expression for the pulse start."""
-
-        if not isinstance(self._start, se_int):
-            self._start = se_int(self._start)["_p" + str(self._id) + "_start"]
-        return self._start
-
-    @property
-    def se_duration(self) -> se_int:
-        """Returns a symbolic expression for the pulse duration."""
-
-        if not isinstance(self._duration, se_int):
-            self._duration = se_int(self._duration)["_p" + str(self._id) + "_duration"]
-        return self._duration
-
-    @property
-    def se_finish(self) -> se_int:
-        """Returns a symbolic expression for the pulse finish."""
-
-        if not isinstance(self._finish, se_int):
-            self._finish = se_int(self._finish)["_p" + str(self._id) + "_finish"]
-        return self._finish
-
-    @property
-    def amplitude(self) -> float:
-        """Returns the amplitude of the pulse.
-
-        Pulse amplitudes are normalised between -1 and 1.
-        """
-
-        return self._amplitude
-
-    @amplitude.setter
-    def amplitude(self, value):
-        """Sets the amplitude of the pulse.
+    The value has to be in the range [10e6 to 300e6].
+    """
+    relative_phase: float
+    """Relative phase of the pulse, in radians."""
+    shape: PulseShape
+    """Pulse shape, as a PulseShape object.
 
-        Args:
-            value (int | float | np.floating): a unitless value between -1 and 1.
-        """
+    See
+    :py: mod:`qibolab.pulses` for list of available shapes.
+    """
+    channel: Optional[str] = None
+    """Channel on which the pulse should be played.
 
-        if isinstance(value, int):
-            value = float(value)
-        if not isinstance(value, (float, np.floating)):
-            raise TypeError(
-                f"amplitude argument type should be float, got {type(value).__name__}"
-            )
-        if not ((value >= -1) & (value <= 1)):
-            raise ValueError(f"amplitude argument must be >= -1 & <= 1, got {value}")
-        if isinstance(value, np.floating):
-            self._amplitude = float(value)
-        elif isinstance(value, float):
-            self._amplitude = value
+    When a sequence of pulses is sent to the platform for execution,
+    each pulse is sent to the instrument responsible for playing pulses
+    the pulse channel. The connection of instruments with channels is
+    defined in the platform runcard.
+    """
+    type: PulseType = PulseType.DRIVE
+    """Pulse type, as an element of PulseType enumeration."""
+    qubit: int = 0
+    """Qubit or coupler addressed by the pulse."""
+    _if: int = 0
+
+    def __post_init__(self):
+        if isinstance(self.type, str):
+            self.type = PulseType(self.type)
+        if isinstance(self.shape, str):
+            self.shape = PulseShape.eval(self.shape)
```

Comment:
What is the reason we expect to see `str` for these two fields? It would be nicer if the entities initializing the pulse conform with the types expected by the Pulse dataclass and we don't do conversion like this here.

Reply:
Because of serialization: it is the chosen way to represent them in a dump.

Instead of nested deserialization (that in my mind is going to come, but it's a bit more complex, because we'll need more tools) we are delegating the whole deserialization responsibility to `Pulse`, including deserializing nested attributes with non-basic types.

Reply:
And serialization is involved because of qibocal runcards?

Reply:
Pulses should not end up in the runcard. Apart from native pulses, which have a completely independent (but somehow redundant) structure.

I believe it is not a bad idea to be able to serialize our objects at any time, but I actually don't remember where it could be used...
(we are not dumping pulse sequences anywhere that I know)

Reply:
To be fair, the origin of this code is to preserve the compatibility with the existing setter, which was even much more verbose:
https://github.com/qiboteam/qibolab/blob/cc8a1b46754f91d3fc80cef00f941575f1f0f223/src/qibolab/pulses.py#L1006-L1011

I'm not sure is used at all (and serialization was more my personal understanding, rather than something that I really know - sorry to realize it in a second stage), but I'd rather keep for this PR, to break the least.
It's simple enough that it will take nothing to drop.

If you want, open an issue to remind about this.

Reply:
I think I mistakenly called config files like [this](https://github.com/qiboteam/qibolab_platforms_qrc/blob/main/iqm5q.yml) a runcard. Since you are preserving previously existing code, then for this PR no need to do further changes. However, in general I would like to get rid of this and actually create pulses with correct types always. Will create a separate issue for that.

Reply:
No mistake: that *is* a runcard (a *platform runcard*, then there *action runcards*, or whatever other name you wish, for Qibocal - it is a mess, and we never found better names).

Actually, the full "runcard" in a sense is the joint product of the Python file and associated YAML, since occasionally some information has been traded from the one side to the other.
However, we usually call "runcard" just the YAML (the separation between the two is that the Python should encode setup specific and immutable information, like connections, while the YAML the parameters that could be calibrated or just measured).